presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/StaticMetastoreLocator.java

```diff
@@ -13,30 +13,36 @@
  */
 package io.prestosql.plugin.hive.metastore.thrift;
 
+import com.google.common.base.Ticker;
 import com.google.common.net.HostAndPort;
+import io.airlift.units.Duration;
 import io.prestosql.plugin.hive.authentication.HiveAuthenticationConfig;
 import io.prestosql.plugin.hive.authentication.HiveAuthenticationConfig.HiveMetastoreAuthenticationType;
+import io.prestosql.plugin.hive.metastore.thrift.FailureAwareThriftMetastoreClient.Callback;
 import org.apache.thrift.TException;
 
 import javax.annotation.Nullable;
 import javax.inject.Inject;
 
 import java.net.URI;
-import java.util.ArrayList;
-import java.util.Collections;
+import java.util.Comparator;
 import java.util.List;
 import java.util.Optional;
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.base.Strings.isNullOrEmpty;
+import static com.google.common.collect.ImmutableList.toImmutableList;
 import static io.prestosql.plugin.hive.metastore.thrift.StaticMetastoreConfig.HIVE_METASTORE_USERNAME;
 import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
 import static java.util.stream.Collectors.toList;
 
 public class StaticMetastoreLocator
         implements MetastoreLocator
 {
-    private final List<HostAndPort> addresses;
+    private final List<Address> addresses;
```

Comment:
Instead of `Address` can you have a map here? I am kind of dislike `Address` bean and it looks not needed.

Reply:
I think introducing `Address` class improve the readability a lot. Otherwise, I need a List<Map.Entry<HostAndPort, Backoff>> in `createMetastoreClient`.

Reply:
Or you can have a list of backoffs

Reply:
I changed to use another list. Please check again. @kokosing